src/main/java/duke/classes/Data.java

```diff
@@ -0,0 +1,130 @@
+package duke.classes;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import duke.exceptions.DukeInvalidTimeException;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.TaskType;
+import duke.tasks.Todo;
+
+/**
+ * Data class that handles the storage and loading of the tasks stored
+ * on the list by writing its contents onto local disk.
+ */
+public class Data {
+    final Path path;
+
+    /**
+     * Constructor that reads file from the path directory, else it creates a file on the directory instead.
+     *
+     * @param path File path for the data stored
+     * @throws IOException For missing files
+     */
+    public Data(String path) throws IOException {
+        //System.out.println("Path being located");
+        this.path = Paths.get("src/main/data/duke.txt").toAbsolutePath();
+        if (Files.notExists(this.path)) {
+            new File(String.valueOf(path)).createNewFile();
+        }
+    }
+
+    /**
+     * Constructor that creates file and directory, if directory is not found.
+     *
+     * @throws IOException For missing files
+     */
+
+    Data() throws IOException {
+        this.path = Path.of("src/main/data/duke.txt");
+        new File("src/main/data").mkdirs();
+        new File(this.path.toString()).createNewFile();
+    }
+
+    /**
+     * Method to read the data stored on local disk.
+     *
+     * @return List of tasks saved
+     * @throws FileNotFoundException For missing files
+     * @throws DukeInvalidTimeException For invalid times saved for event/deadline events
+     * @throws ArrayIndexOutOfBoundsException For incorrectly typed descriptions for deadline/event tasks
+     */
+    public List<Task> loadData() throws FileNotFoundException, DukeInvalidTimeException,
+            ArrayIndexOutOfBoundsException {
+        List<Task> todoList = new ArrayList<>();
+        Scanner scanner = new Scanner(path.toFile());
+
+        for (int i = 1; scanner.hasNextLine(); i++) {
+            Task currTask = null;
+            String[] curr = scanner.nextLine().split("---");
```

Comment:
Might be better to name it such that it is a plural since it is an array. Perhaps 'currSubStrings'?

Reply:
Thanks a lot of the helpful feedback! Will look towards implementing it into my code!